fair_mango/metrics/superset.py

```diff
@@ -0,0 +1,365 @@
+        ...         'race': ['white', 'white', 'black', 'black', 'white', 'white'],
+        ...         'real_churn': [1,0,1,0,0,1],
+        ...         'pred_churn': [0,0,1,0,0,1]
+        ... })
+        >>> result = super_set_performance_metrics(
+        ...     data=df,
+        ...     sensitive=['gender', 'race'],
+        ...     real_target=['real_churn'],
+        ...     predicted_target=['pred_churn'],
+        ... )
+        >>> result
+        [
+            {'sensitive': ('gender',),
+            'result': (['real_churn'],
+            [
+                {
+                    'sensitive': array(['male'], dtype=object),
+                    'selection_rate_in_data': array(0.5),
+                    'selection_rate_in_predictions': array(0.25),
+                    'accuracy': [0.75],
+                    'balanced accuracy': [0.75],
+                    'precision': [1.0],
+                    'recall': [0.5],
+                    'f1-score': [0.6666666666666666],
+                    'false_negative_rate': [0.5],
+                    'false_positive_rate': [0.0],
+                    'true_negative_rate': [1.0],
+                    'true_positive_rate': [0.5]
+                },
+                {
+                    'sensitive': array(['female'], dtype=object),
+                    ...
+                    'f1-score': [0.0],
+                    'false_negative_rate': [1.0],
+                    'false_positive_rate': [0.0],
+                    'true_negative_rate': [1.0],
+                    'true_positive_rate': [0.0]
+                }
+            ]
+            )
+            }
+        ]
+        """
+        results = []
+
+        for pair in self.pairs:
+            concatenated_results = SelectionRate(
+                data=self.data,
+                use_y_true=True,
+                sensitive=list(pair),
+                real_target=self.real_target,
+                predicted_target=self.predicted_target,
+                positive_target=self.positive_target,
+                label="selection_rate_in_data",
+            )()
+
+            for metric in self.metrics:
+                if metric is SelectionRate:
+                    result = SelectionRate(
+                        data=self.data,
+                        use_y_true=False,
+                        sensitive=list(pair),
+                        real_target=self.real_target,
+                        predicted_target=self.predicted_target,
+                        positive_target=self.positive_target,
+                        label="selection_rate_in_predictions",
+                    )()[1]
+                else:
+                    result = metric(
+                        data=self.data,
+                        sensitive=list(pair),
+                        real_target=self.real_target,
+                        predicted_target=self.predicted_target,
+                        positive_target=self.positive_target,
+                    )()[1]
```

Comment:
it looks very similar but tbh, I don't think there is an elegant way to remove an argument except by using dictionnaries

Reply:
I don't see a better way to do this.